[TensorPipe] Re-enable RPC tests

lw opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39441 [TensorPipe] Re-enable dist optimizer tests
* #39440 [TensorPipe] Re-enable dist autograd tests
* **#39406 [TensorPipe] Re-enable RPC tests**

For now, just the RPC test (no dist autograd or dist optimizer).

I removed the skipping decorator from all the tests except those that explicitly use the ProcessGroup options.

Includes #39027.

Differential Revision: [D21716069](https://our.internmc.facebook.com/intern/diff/D21716069/)

lw committed:
[TensorPipe] Re-enable RPC tests

For now, just the RPC test (no dist autograd or dist optimizer).

I removed the skipping decorator from all the tests except those that explicitly use the ProcessGroup options.

Includes #39027.

Differential Revision: [D21716069](https://our.internmc.facebook.com/intern/diff/D21716069/)

[ghstack-poisoned]

lw commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -2429,8 +2403,9 @@ def _create_rref(self):
     @dist_init
-    @_skip_if_tensorpipe_agent
     def test_user_rrefs_confirmed(self):
+        print(os.getpid())
+        # time.sleep(20)
```
Oups, these slipped in from my debugging sessions, I'll remove them (and the import above) but I first want to wait for the CI to finish.

mrshenli commented on test/run_test.py:
```diff
@@ -72,6 +72,9 @@
     'distributed/rpc/faulty_agent/test_dist_autograd_spawn',
     'distributed/rpc/faulty_agent/test_rpc_spawn',
     'distributed/rpc/jit/test_dist_autograd_spawn',
+    # 'distributed/rpc/tensorpipe/test_dist_autograd_spawn',
```
Let's remove these commented-out lines from this PR. We can add them back when we are ready.

mrshenli reviewed (approved):
Stamp to unblock, but please wait for tests and remove commented-out lines. Thanks!

osalpekar commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -2429,8 +2403,9 @@ def _create_rref(self):
     @dist_init
-    @_skip_if_tensorpipe_agent
     def test_user_rrefs_confirmed(self):
+        print(os.getpid())
```
Are these debugging artifacts that we should remove?

osalpekar commented on torch/testing/_internal/distributed/rpc/rpc_test.py:
```diff
@@ -1,4 +1,5 @@
 import concurrent.futures
+import os
```
Do we need this import? (apart from the os.getpid call below?)

osalpekar reviewed (approved):
Looks good - what do we plan on doing with #39027?

lw committed:
Update on "[TensorPipe] Re-enable RPC tests"


For now, just the RPC test (no dist autograd or dist optimizer).

I removed the skipping decorator from all the tests except those that explicitly use the ProcessGroup options.

Includes #39027.

Differential Revision: [D21716069](https://our.internmc.facebook.com/intern/diff/D21716069/)

[ghstack-poisoned]
